mbuild/compound.py

```diff
@@ -26,13 +24,13 @@
 from mbuild.formats.gsdwriter import write_gsd
 from mbuild.formats.par_writer import write_par
 from mbuild.periodic_kdtree import PeriodicCKDTree
-from mbuild.utils.io import run_from_ipython, import_, has_networkx
+from mbuild.utils.io import run_from_ipython, import_, has_networkx, has_mdtraj
 from mbuild.utils.jsutils import overwrite_nglview_default
 from mbuild.coordinate_transform import _translate, _rotate
 
 
 def load(filename_or_object, relative_to_module=None, compound=None, coords_only=False,
-         rigid=False, use_parmed=False, smiles=False, 
+         rigid=False, use_parmed=True, smiles=False, 
```

Comment:
We want mdtraj to load hoomdxml files, but it fails at L149 since it is trying to load it using parmed. Just curious, why did you make this true?

Reply:
I neglected this case and assumed falling back to parmed would work for all formats

Reply:
smiles strings don't work when this is true also.

Reply:
I need to put some more thought into this, probably worth of a discussion, but I have wondered if instead of defaulting to MDTraj and allowing ParmEd to override that default with a boolean flag, we should have a `backend='foo'` argument. It could default to ParmEd but optionally take in MDTraj, Topology, OpenBabel, etc.

Reply:
Yeah. Ideally the default behavior would be to try all of the backends and then error out if none of them work. A `backend` arg is a good idea for the case where you know what loader you want to use and want to skip trying the other loaders.

Reply:
agreed, I like the idea of a `backend` argument. Fits the way we handle `Compound.visualize` too.